File: C#/0199. Binary Tree Right Side View.cs
/**
 * Definition for a binary tree node.
 * public class TreeNode {
 *     public int val;
 *     public TreeNode left;
 *     public TreeNode right;
 *     public TreeNode(int x) { val = x; }
 * }
 */
public class Solution {
    public IList<int> RightSideView(TreeNode root) {
        var result=new List<int>();
        if(root==null)
            return result;
        var rootQueue=new Queue<TreeNode>();
        rootQueue.Enqueue(root);
        while(rootQueue.Count()!=0){
            int lenRootQueue=rootQueue.Count();
            for(int i=0;i<lenRootQueue;i++){
                var curRoot=rootQueue.Dequeue();
                if(curRoot.left!=null){
                    rootQueue.Enqueue(curRoot.left);
                }
                if(curRoot.right!=null){
                    rootQueue.Enqueue(curRoot.right);
                }
                if(i==lenRootQueue-1){
                    result.Add(curRoot.val);
                }
            }
        }
        return result;
    }
}

File: archives/20190519python/0199. Binary Tree Right Side View.py
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, x):
#         self.val = x
#         self.left = None
#         self.right = None

class Solution:
    def rightSideView(self, root: TreeNode) -> List[int]:
        if not root:
            return []
        rep=[]
        root_stack=[root]
        while root_stack:
            rep.append(root_stack[-1].val)
            len_root_stack=len(root_stack)
            for k in range(len_root_stack):
                cur_root=root_stack.pop(0)
                if cur_root.left:
                    root_stack.append(cur_root.left)
                if cur_root.right:
                    root_stack.append(cur_root.right)
        return rep
